src/components/RequestRepository/AnnualEmail.tsx

```diff
@@ -198,62 +207,91 @@ const AnnualEmail = () => {
-                  data-testid={`checkbox-${key}`}
+        {loading && (
+          <div className="display-flex flex-align-center flex-justify-space-between">
+            <Icon.Autorenew className="text-base-dark" />
+            <span className="margin-left-1 text-base-dark text-italic">
+              {t('general:loadingData')}
+            </span>
+          </div>
+        )}
+        {error && (
+          <Alert type="error" isClosable={false}>
+            {t('home:adminHome.GRT.requesterUpdateEmail.modal.error')}
+          </Alert>
+        )}
+
+        {!loading && !error && (
+          <>
+            <Form onSubmit={handleSubmit(onSubmit)} className="maxw-none">
+              {typedEntries(list).map(([key, statusLabel]) => (
+                <Controller
+                  key={key}
+                  name={key}
+                  control={control}
+                  render={({ field: { ref, ...field } }) => (
+                    <CheckboxField
+                      {...field}
+                      id={key as string}
+                      value={key as string}
+                      label={statusLabel}
+                      data-testid={`checkbox-${key}`}
+                    />
+                  )}
                 />
+              ))}
+
+              {warning && (
+                <Alert slim type="warning">
+                  {t('home:adminHome.GRT.requesterUpdateEmail.modal.noEmail')}
+                </Alert>
               )}
-            />
-          ))}
-
-          {warning && (
-            <Alert slim type="warning">
-              {t('home:adminHome.GRT.requesterUpdateEmail.modal.noEmail')}
-            </Alert>
-          )}
-
-          <ModalFooter>
-            <div className="display-flex flex-gap-3">
-              <Button
-                type="submit"
-                disabled={!hasSelected}
-                className="margin-y-0"
-              >
-                {t(
-                  'home:adminHome.GRT.requesterUpdateEmail.modal.openEmailButton'
-                )}
-              </Button>
-              <Button
-                type="button"
-                onClick={onCopy}
-                disabled={!hasSelected}
-                unstyled
-                className="margin-y-0"
-              >
-                {t(
-                  'home:adminHome.GRT.requesterUpdateEmail.modal.copyEmailButton'
-                )}
-              </Button>
-            </div>
-          </ModalFooter>
-        </Form>
+
+              <ModalFooter>
+                <div className="display-flex flex-gap-3">
+                  <Button
+                    type="submit"
+                    disabled={!hasSelected}
+                    className="margin-y-0"
+                  >
+                    {t(
+                      'home:adminHome.GRT.requesterUpdateEmail.modal.openEmailButton'
+                    )}
+                  </Button>
+                  {emailsCopied ? (
```

Comment:
Can you reset `emailsCopied` when the checkboxes are updated? Right now you need to close and reopen the modal if you want to update your selection and copy the new emails.